### DISCUSSION TITLE
Major RTPS Version not supported [13587]

### DISCUSSION THREAD

**Discussion by ghost** [eProsima/Fast-DDS#2433] [Ideas]
When receiving messages in a Java-based [KIARA application](https://fiware-middleware-kiara.readthedocs.io/en/latest/manuals/User_and_Programmer_Guide.html) from a C-based Fast DDS publisher, the software reports that the RTPS version is not supported. This information is helpful in that it shows the problem, yet could be improved by being more specific as to the technical requirements not being met. The specificity should include both the expected value and the actual value. In the source code for log strings throughout this ticket, one or both are missing.

##  Actual

```
[main] INFO org.fiware.kiara.ps.rtps.participant.RTPSParticipant - Reader creation finished successfully
[main] INFO org.fiware.kiara.ps.participant.Participant - Subscriber 01 0F 3B 82 00 00 70 44 00 00 00 00  - 00 00 01 04  created in topic Phy_Rx_Packet
[] WARN org.fiware.kiara.ps.rtps.messages.MessageReceiver - Major RTPS Version not supported
```

## Expected

```
[main] INFO org.fiware.kiara.ps.rtps.participant.RTPSParticipant - Reader creation finished successfully
[main] INFO org.fiware.kiara.ps.participant.Participant - Subscriber 01 0F 3B 82 00 00 70 44 00 00 00 00  - 00 00 01 04  created in topic Phy_Rx_Packet
[] WARN org.fiware.kiara.ps.rtps.messages.MessageReceiver - Major RTPS Version not supported (expected >= 3, actual = 2)
```

## Technical

Change the following line to log the incident major version number received and the requisite major version number(s):

https://github.com/eProsima/Fast-DDS/blob/1f9247714c145bc601c12eae65bcc4d236514404/src/cpp/rtps/messages/MessageReceiver.cpp#L538

## Additional

There are other places in the code base where similar messages could be improved by providing expected and actual values. For example, in the same file, we see:

    logWarning(RTPS_MSG_IN, IDSTRING "SubmessageHeader too short");

This log message could be augmented with superpowers by providing the expected/actual values:

    SubmessageHeader byte count too short (expected > 3, actual = 0)

Others:

    logInfo(RTPS_MSG_IN, IDSTRING "Too short submessage received, ignoring");
    logWarning(RTPS_MSG_IN, IDSTRING "Invalid message received, bad sequence Number");
    logWarning(RTPS_MSG_IN, IDSTRING "Ignoring Serialized Payload for too large key-only data (" <<

Nit: Usually log messages don't reference the first person singular:

    logInfo(RTPS_MSG_IN, IDSTRING "Acknack msg to UNKNOWN writer (I looked through "

**Comment by JLBuenoLopez** [comment]
Hi @DaveJarvis,

First of all, sorry for the late reply.

Thanks for your suggestion. I have moved the issue to the Ideas discussion section where we intend to keep the user suggestions to improve Fast DDS but that are not properly a bug or malfunction.
